rpc-rs/src/timestamp.rs

```diff
@@ -28,11 +29,25 @@ pub struct Timestamp {
 }
 
 impl Timestamp {
+    /// Constructs a timestamp, with parameter validation
+    pub fn new(sec: i64, nsec: u32) -> Result<Self, &'static str> {
+        if sec > 0 && (sec < i64::MAX / 1_000_000_001) && nsec < 1_000_000_000 {
```

Comment:
I don't really understand what this asserts, `sec < ~ 9 billion`? nsec < 1 billion?

Reply:
Those two conditions ensure that you can calculate absolute time in  nanoseconds in an i64 with sec * 1000000000 + nsec